src/svc_xprt.c

```diff
@@ -173,69 +198,44 @@ svc_xprt_lookup(int fd, svc_xprt_setup_t setup)
 	return (xprt);
 }
 
-SVCXPRT *
-svc_xprt_get(int fd)
-{
-	struct rpc_svcxprt sk;
-	struct rbtree_x_part *t;
-	struct opr_rbtree_node *nv;
-	SVCXPRT *srec = NULL;
-
-	if (svc_xprt_init_failure())
-		return (NULL);
-
-	sk.xp_fd = fd;
-	t = rbtx_partition_of_scalar(&svc_xprt_fd.xt, fd);
-
-	rwlock_rdlock(&t->lock);
-	nv = opr_rbtree_lookup(&t->t, &sk.xp_fd_node);
-	rwlock_unlock(&t->lock);
-
-	/* XXX safe, even if tree is reorganizing */
-	if (nv)
-		srec = opr_containerof(nv, struct rpc_svcxprt, xp_fd_node);
-
-	return (srec);
-}
-
 /**
  * Clear an xprt
  *
  * @note Locking
- * - xprt is locked, unless RPC_DPLX_FLAG_LOCKED is passed
- * - xprt is unlocked if RPC_DPLX_FLAG_UNLOCK is passed, otherwise it is
- *   returned locked
+ * - xprt is locked, unless SVC_XPRT_FLAG_BLOCKED
+ * - xprt is unlocked, unless SVC_XPRT_FLAG_BLOCKED
+ *   otherwise it is returned locked
```

Comment:
Can a receive  happen in the middle of this function?  If not, I don't see how it can ever return locked.  This doesn't seem to be called with it locked.